tls/s2n_client_cert_verify.c

```diff
@@ -33,27 +33,32 @@ int s2n_client_cert_verify_recv(struct s2n_connection *conn)
 
     struct s2n_stuffer *in = &conn->handshake.io;
 
-    POSIX_GUARD_RESULT(s2n_signature_algorithm_recv(conn, in));
-    const struct s2n_signature_scheme *chosen_sig_scheme = conn->handshake_params.client_cert_sig_scheme;
-    POSIX_ENSURE_REF(chosen_sig_scheme);
-
-    uint16_t signature_size = 0;
-    struct s2n_blob signature = { 0 };
-    POSIX_GUARD(s2n_stuffer_read_uint16(in, &signature_size));
-    signature.size = signature_size;
-    signature.data = s2n_stuffer_raw_read(in, signature.size);
-    POSIX_ENSURE_REF(signature.data);
-
-    /* Use a copy of the hash state since the verify digest computation may modify the running hash state we need later. */
-    struct s2n_hash_state *hash_state = &hashes->hash_workspace;
-    POSIX_GUARD_RESULT(s2n_handshake_copy_hash_state(conn, chosen_sig_scheme->hash_alg, hash_state));
-
-    /* Verify the signature */
-    POSIX_GUARD(s2n_pkey_verify(&conn->handshake_params.client_public_key, chosen_sig_scheme->sig_alg, hash_state, &signature));
+    if (conn->op.async_op_state == S2N_ASYNC_NOT_INVOKED) {
```

Comment:
I don't like that since conn->op will be reused for multiple async operations, we're relying on `s2n_async_op_wipe()` to be properly called in order to prevent previous operation invocations from messing with subsequent ones. For example, if an operation was invoked before this that forgot to call `s2n_async_op_wipe()`, `async_op_state` would still be set to `S2N_ASYNC_COMPLETE` and the call to `s2n_async_pkey_verify` would be skipped.

I'm not sure of the best way to prevent this. Could we maybe add a check when `async_op_state == S2N_ASYNC_COMPLETE` to make sure the op type is `S2N_ASYNC_PKEY_VERIFY`?